market-generator/src/test/java/market/MarketSubscriptionManagerTest.java

```diff
@@ -0,0 +1,159 @@
+package market;
+
+import com.market.MarketDataRequest;
+import com.market.TickResponse;
+import com.market.banica.generator.exception.NoSuchGoodException;
+import com.market.banica.generator.service.MarketSubscriptionManager;
+import io.grpc.stub.StreamObserver;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.HashSet;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class MarketSubscriptionManagerTest {
```

Comment:
please add Arrange/Act/Assert or given/when/then for each of your tests to be more clear and understandable and follow this for all the tests